deepspeed/runtime/engine.py

```diff
@@ -2229,8 +2229,22 @@ def _get_all_zero_checkpoints(self, load_dir, tag):
             return None
 
         zero_sd_list = []
-        for ckpt_name in zero_ckpt_names:
-            zero_sd_list.append(torch.load(ckpt_name, map_location='cpu'))
+        for i, ckpt_name in enumerate(zero_ckpt_names):
+            # Fully load state for current rank
+            # TODO: remove assumption on zero_sd_list[0] in zero load_state_dict and use rank instead
+            if self.zero_elastic_checkpoint() or i == 0 or dist.get_rank(
+                    group=self.optimizer.dp_process_group) == i:
+                zero_sd_list.append(torch.load(ckpt_name, map_location='cpu'))
+            else:
+                _state = torch.load(ckpt_name, map_location='cpu')
+                # Extract fp32 groups only, otherwise throw away to prevent unnecessary CPU memory overheads
+                _state = {
+                    'optimizer_state_dict': {
```

Comment:
Can we replace the hardcoded constants `optimizer_state_dict` and `single_partition_of_fp32_groups` with variables imported from stage2.py to ease code maintenance?